DataBiosphere/terra-workspace-data-service
# AJ-1011: snapshot import via Quartz

connects the dots and performs async snapshot import via Quartz. Well, as of this PR the import fails because we're not properly passing a token. But you can see that it's trying, and token work will be a follow-on PR ([Slack thread](https://broadinstitute.slack.com/archives/C02SJQ0FKHQ/p1697205316686489))

To test this locally:
1. Export a snapshot from TDR via the [exportSnapshot API](https://jade.datarepo-dev.broadinstitute.org/swagger-ui.html#/snapshots/exportSnapshot) and copy the url to the exported manifest
2. Call WDS's `importV1` API with a payload of `{"type": "TDRMANIFEST", "url": "url-to-manifest-from-previous-step"}` and make note of the jobId it returns
3. Call WDS's `jobStatusV1` API, passing the jobId from the previous step
4. View the WDS logs to see progress of the job

For future PRs:
* Properly pass an auth token from the Quartz job to the WSM and TDR API calls
* return some kind of response payload from the import, such as the number of rows affected
* Update the OpenAPI spec to define the expected response payloads
* always can use more test coverage
